### PR TITLE
feat(compile): Noir std lib embedded

### PR DESCRIPTION
**Pull request by kobyhallx**
# Related issue(s)

Relates to #951

# Description

It's beneficial to embed std lib into distributed binary:
1. Source resolution does not concern itself with differences between execution environments (eg. native vs browser)
2. Source resolution does not concern itself with differences between systems and default file locations
3. Source resolution does not concern itself with differences between systems and file permissions
4. Distribution of artifacts becomes more straightforward with less maintenance needed for installation scripts 

## Summary of changes

1. `noir_stdlib/src` contents get embedded as `StdLibAssets` in `crates/fm/src/file_reader.rs`
2. `StdLibAssets` is asked for source upfront during `read_file_to_string` routine
3. `crates/nargo/src/cli/mod.rs` understands that `StdLibAssets` are embedded with `std/` prefix and adds such search path for `create_non_local_crate` routine.

## Dependency additions / changes

Adds `rust-embed = "6.6.0"`

## Test additions / changes

N/A

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

## Documentation needs
- [x] This PR requires documentation updates when merged:
> Std lib no longer needs to be present on a host file system, and libraries do not need to provide the source in alternative ways, ie. source resolution in browser

# Additional context

n/a

### CODE REVIEW

File: crates/wasm/src/lib.rs [hunk 1]
```diff
@@ -54,6 +54,9 @@ pub fn compile(args: JsValue) -> JsValue {
 
     driver.create_local_crate(path, CrateType::Binary);
 
+    // We are always adding std lib implicitly. It comes bundled with binary.
+    add_noir_lib(&mut driver, (&"std").to_string());
```
Comment (TomAFrench) [inline crates/wasm/src/lib.rs:58]:
```suggestion
    add_noir_lib(&mut driver, "std");
```
We can change `add_noir_lib` to take a `&str` which should make things clearer.

**Comment by TomAFrench** [comment]
Now that we'll always have the stdlib available, should we not embed the addition of the stdlib onto the end of `Driver::resolve_root_config`?

**Comment by kobyhallx** [comment]
> ... in any case we should be able to abstract away the addition of the stdlib into the dependency graph.
agreed, should be refactored to more common path at later stage

**Review by TomAFrench — APPROVED** [review]
Looks good. I've run a quick test locally just to confirm that we're actually pulling the stdlib from an embedded version and all looks good.

**Pull request merged**